cdap-docs/reference-manual/source/java-client-api.rst

```diff
@@ -335,39 +327,20 @@ QueryClient
   // Construct the client used to interact with CDAP
   QueryClient queryClient = new QueryClient(clientConfig);
 
-  //
   // Perform an ad-hoc query using the Purchase example
-  //
-  String query = "SELECT * FROM dataset_history WHERE customer IN ('Alice','Bob')"
-  QueryHandle queryHandle = queryClient.execute(query);
-  QueryStatus status = new QueryStatus(null, false);
-
-  while (QueryStatus.OpStatus.RUNNING == status.getStatus() ||
-         QueryStatus.OpStatus.INITIALIZED == status.getStatus() ||
-         QueryStatus.OpStatus.PENDING == status.getStatus()) {
-    Thread.sleep(1000);
-    status = queryClient.getStatus(queryHandle);
-  }
-
-  if (status.hasResults()) {
-    // Get first 20 results
-    List<QueryResult> results = queryClient.getResults(queryHandle, 20);
-    // Fetch schema
-    List<ColumnDesc> schema = queryClient.getSchema(queryHandle);
-    String[] header = new String[schema.size()];
-    for (int i = 0; i < header.length; i++) {
-      ColumnDesc column = schema.get(i);
-      // Hive columns start at 1
-      int index = column.getPosition() - 1;
-      header[index] = column.getName() + ": " + column.getType();
-    }
+  ListenableFuture<ExploreExecutionResult> resultFuture = queryClient.execute(Id.Namespace.DEFAULT, "SELECT * FROM dataset_history WHERE customer IN ('Alice','Bob')");
+  ExploreExecutionResult results = resultFuture.get();
+
+  // Fetch schema
+  List<ColumnDesc> schema = results.getResultSchema();
+  String[] header = new String[schema.size()];
+  for (int i = 0; i < header.length; i++) {
+  ColumnDesc column = schema.get(i);
```

Comment:
Indentation isn't preserved unfortunately. But yes, we can do begin/endIndent().